volatility3/framework/plugins/windows/skeleton_key_check.py

```diff
@@ -0,0 +1,578 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+# This module attempts to locate skeleton-key like function hooks.
+# It does this by locating the CSystems array through a variety of methods,
+# and then validating the entry for RC4 HMAC (0x17 / 23)
+#
+# For a thorough walkthrough on how the R&D was performed to develop this plugin, 
+# please see our blogpost here:
+#
+# <insert blog URL once published>
+
+import logging, io
+
+from typing import Iterable, Tuple
+
+from volatility3.framework.symbols.windows import pdbutil
+from volatility3.framework import interfaces, symbols, exceptions
+from volatility3.framework import renderers, constants
+from volatility3.framework.layers import scanners
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.symbols import intermed
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import pslist, vadinfo
+
+from volatility3.framework.symbols.windows.extensions import pe
+
+try:
+    import capstone
+    has_capstone = True
+except ImportError:
+    has_capstone = False
+
+try:
+    import pefile
+    has_pefile = True
+except ImportError:
+    has_pefile = False
+
+vollog = logging.getLogger(__name__)
+
+class Skeleton_Key_Check(interfaces.plugins.PluginInterface):
+    """Lists process memory ranges that potentially contain injected code."""
+
+    _required_framework_version = (1, 0, 0)
```

Comment:
I took a look through and the only function that I see being re-usable (not skeleton key specific) is __get_rip_relative_target_. This is actually usable by plugins for any OS analyzing x86_64 targets. Would you like me to version based on that one? Or should that function go in API somewhere instead?

Reply:
Hmmm, it's a pretty straight forward calculation, but there's enough that could go wrong in there we probably do want it somewhere common.  I don't think people would think to look in the `skeleton_key_check` file when they wanted things related to disassembly, but I'd like to avoid adding libraries of useful functions to core (since it doesn't show external developers how they can do something similar), so it might be worthwhile starting a `disassembly.py` that contains a `VersionableInterface` derived object (doesn't have to be a plugin, can just be a collection of class/static methods) which people can then pull in using the requirements mechanism.  In that case, the skeleton key plugin doesn't need versioning if it doesn't expose anything...

Reply:
i like the disassembly.py idea. Which folder would that go in since its usable across all OSes?

Reply:
Hmmm, I'd have said `symbols` and then either directly in there, or we can start a `generic` directory for all of them (similar to volshell's terminology), but that then prevents people from overriding it, so it might be better if it lives directly under plugins?  I might ask for opinions at the next meeting (remind me if I forget)...  5:)